src/.vuepress/markdown/link.js

```diff
@@ -0,0 +1,100 @@
+// markdown-it plugin for:
+// 1. adding target="_blank" to external links
+// 2. converting internal links to <router-link>
+
+const url = require('url')
+
+const indexRE = /(^|.*\/)(index|readme).md(#?.*)$/i
+
+module.exports = (configuration) => {
+  return (md) => {
+    let hasOpenRouterLink = false
+    let hasOpenExternalLink = false
+
+    md.renderer.rules.link_open = (tokens, idx, options, env, self) => {
+    const { base } = configuration;
+    const startsWithBaseRE = new RegExp(`^${base}.*`)
+    const internalRE = /(\/|\.md|\.html)(#.*)?$/
+    const { relativePath } = env
+      const token = tokens[idx]
+      const hrefIndex = token.attrIndex('href')
+      if (hrefIndex >= 0) {
+        const link = token.attrs[hrefIndex]
+        const href = link[1]
+        const isExternal = /^https?:/.test(href)
+        const isInternal = internalRE.test(href)
+        const startsWithBaseURL = startsWithBaseRE.test(href)
```

Comment:
couldn't this be `href.startsWith(base)` ?

Reply:
yes, it is equivalent